modules/objdetect/src/face_detect.cpp

```diff
@@ -0,0 +1,290 @@
+    }
+
+    int detect(InputArray input_image, OutputArray faces) override
+    {
+        // TODO: more checkings should be done?
+        if (input_image.empty())
+        {
+            return 0;
+        }
+
+        // Build blob from input image
+        Mat input_blob = dnn::blobFromImage(input_image);
+
+        // Forward
+        std::vector<String> output_names = { "loc", "conf", "iou" };
+        std::vector<Mat> output_blobs;
+        net.setInput(input_blob);
+        net.forward(output_blobs, output_names);
+
+        // Post process
+        Mat results = postProcess(output_blobs);
+        results.convertTo(faces, CV_32FC1);
+        return 1;
+    }
+private:
+    void generatePriors()
+    {
+        // Calculate shapes of different scales according to the shape of input image
+        Size feature_map_2nd = {
+            int(int((img_w+1)/2)/2), int(int((img_h+1)/2)/2)
+        };
+        Size feature_map_3rd = {
+            int(feature_map_2nd.width/2), int(feature_map_2nd.height/2)
+        };
+        Size feature_map_4th = {
+            int(feature_map_3rd.width/2), int(feature_map_3rd.height/2)
+        };
+        Size feature_map_5th = {
+            int(feature_map_4th.width/2), int(feature_map_4th.height/2)
+        };
+        Size feature_map_6th = {
+            int(feature_map_5th.width/2), int(feature_map_5th.height/2)
+        };
+
+        std::vector<Size> feature_map_sizes;
+        feature_map_sizes.push_back(feature_map_3rd);
+        feature_map_sizes.push_back(feature_map_4th);
+        feature_map_sizes.push_back(feature_map_5th);
+        feature_map_sizes.push_back(feature_map_6th);
+
+        // Fixed params for generating priors
+        const std::vector<std::vector<float>> min_sizes = {
+            {10.0f,  16.0f,  24.0f},
+            {32.0f,  48.0f},
+            {64.0f,  96.0f},
+            {128.0f, 192.0f, 256.0f}
+        };
+        const std::vector<int> steps = { 8, 16, 32, 64 };
+
+        // Generate priors
+        priors.clear();
+        for (size_t i = 0; i < feature_map_sizes.size(); ++i)
+        {
+            Size feature_map_size = feature_map_sizes[i];
+            std::vector<float> min_size = min_sizes[i];
+
+            for (int _h = 0; _h < feature_map_size.height; ++_h)
+            {
+                for (int _w = 0; _w < feature_map_size.width; ++_w)
+                {
+                    for (size_t j = 0; j < min_size.size(); ++j)
+                    {
+                        float s_kx = min_size[j] / img_w;
+                        float s_ky = min_size[j] / img_h;
+
+                        float cx = (_w + 0.5) * steps[i] / img_w;
+                        float cy = (_h + 0.5) * steps[i] / img_h;
+
+                        Rect2f prior = { cx, cy, s_kx, s_ky };
+                        priors.push_back(prior);
+                    }
+                }
+            }
+        }
+    }
+
+    Mat postProcess(const std::vector<Mat>& output_blobs)
```

Comment:
Please avoid using of generic `FaceDetector` name for such specialized implementation. Add some suffix, e.g. from article or model name -`FaceDetectorXYZ`.

Reply:
Our original goal was to create a DNN-based face detection API that can use our model at least, and replace the traditional Haar-based one. Maybe we can make it more universal for different models.

Reply:
Done renaming for `FaceDetectorYN` at https://github.com/opencv/opencv/pull/20422/commits/541ed921c55b1325ece074a2b6725ce88968c0eb

Reply:
> FaceRecognizer

Yes, there is the same note about `FaceRecognizer`.